gogen/goelements.go

```diff
@@ -292,18 +291,18 @@ func (s *GoLangMapper) PackageName(*yang.Entry, genutil.CompressBehaviour, bool)
 // The skipEnumDedup argument specifies whether leaves of type enumeration that are
 // used more than once in the schema should share a common type. By default, a single
 // type for each leaf is created.
-func (s *GoLangMapper) yangTypeToGoType(args resolveTypeArgs, compressOCPaths, skipEnumDedup, shortenEnumLeafNames, useDefiningModuleForTypedefEnumNames bool, enumOrgPrefixesToTrim []string) (*MappedType, error) {
+func (s *GoLangMapper) yangTypeToGoType(args resolveTypeArgs, compressOCPaths, skipEnumDedup, shortenEnumLeafNames, useDefiningModuleForTypedefEnumNames bool, enumOrgPrefixesToTrim []string) (*ygen.MappedType, error) {
 	defVal := genutil.TypeDefaultValue(args.yangType)
 	// Handle the case of a typedef which is actually an enumeration.
-	typedefName, _, err := s.EnumeratedTypedefTypeName(args, goEnumPrefix, false, useDefiningModuleForTypedefEnumNames)
+	typedefName, _, err := s.EnumeratedTypedefTypeName(args.yangType, args.contextEntry, goEnumPrefix, false, useDefiningModuleForTypedefEnumNames)
```

Comment:
Could you explain this change? It's not immediately obvious to me what happened here -- we changed the arguments to `EnumeratedTypedefTypeName` (since I think `args.contextEntry` is a `yang.Entry` whereas `goEnumPrefix` is a string).

Reply:
So I took away the `resolveTypeArgs` argument type and expanded it out into the two parameters at the beginning; otherwise `resolveTypeArgs` would need to be exported, and since this is the only exported function where this is needed I judged that introducing another exported type (likely residing in `genutil`) to the user is not worth the overhead of changing all the existing usages.

Reply:
Ah, I see -- I had thought that the move there was mechnical. Great, this is fine with me.